### PR TITLE
[Added Features] Theme/Cache/Render/GUI

### PR DESCRIPTION
**Pull request by ITZVGcGPmO**
added:
- additional theme support
- dark theme
- theme cycle button
- version display now links to github
- links to all the places in which skin data is uploaded
- show player's skull instead of generic icon
- player's info functions as a skin download
- preview player's skin on page load
- widened url input for longer urls
- url input updates preview more often
- skin renderer, modified version of https://github.com/Gyzie/php-Minecraft-3D-Skin-Renderer
- skin cache, so mojang isn't polled every time a skin is fetched
- cache system can hold new and interesting super cool awesome stuff™
- you can set a default theme by setting 'deftheme' to 'light' or 'dark' in config.nogit.php (like how 'version' is 1.6) 
- legacy config support(and update system)
- `authentication limit` limits by ip and, as a backup, by username
- index.php under `themes` lists all themes in json

changed:
- text formatting in places, improved it has.
- reload page on skin upload success/theme toggle
- styles.css is now known as light.css
- yml files are parsed more reliably
- config file writing is now simpler
- default theme can be chosen during installation
- experimental history functionality has been _slightly_ improved
- release version is now defined in only one place
- version number to 1.7
- `authentication security` is now known as `authentication limit`
- auth errors use codes 401, 429
- file inputs had their filetype selectors changed slightly

removed:
- buttons in popup window

### CODE REVIEW

**Comment by riflowth** [comment]
Thank you very much for many features! 👍 😄

File: index.php [hunk 1]
```diff
@@ -54,7 +56,7 @@ function setCookie(cname, cvalue) {
                     <?php if($config['authme']['enabled'] == true && !empty($_SESSION['username'])){ 
                       $SkullURL = '/skin/?vr=0&hr=0&headOnly=true&ratio=4&user='.$_SESSION['username'];
                       echo '<a class="skinDownload" href="/skin/?format=raw&dl=true&user='.$_SESSION['username'].
-                      '"><img class="skinDownload" src="'.$SkullURL.'">    '.htmlspecialchars($_SESSION['username'], ENT_QUOTES).'</a>'; ?>
+                      '"><img class="skinDownload" style="max-height:29px!important;" src="'.$SkullURL.'">    '.htmlspecialchars($_SESSION['username'], ENT_QUOTES).'</a>'; ?>
```
Comment (ITZVGcGPmO) [inline index.php]: accidental css addition :wink:

**Comment by riflowth** [comment]
@ITZVGcGPmO How do I contact you?

**Comment by ITZVGcGPmO** [comment]
i have a discord tag of 087t34w#7838

**Comment by ITZVGcGPmO** [comment]
@riflowth i have a discord tag of 087t34w#7838

**Comment by riflowth** [comment]
You are **AWESOME** 😄 👍 🥇 Thank you so much+++ for tons of commit 👍

**Comment by ITZVGcGPmO** [comment]
it should be ready for release after i check the mattress for bugs

**Comment by riflowth** [comment]
Very awesome pull request 👍 👍 👍 👍 👍 👍 I really love it ❤️

**Comment by ITZVGcGPmO** [comment]
I have exterminated some bugs, and can't found any more. Please tell me if you find any :smile:

**Pull request merged**